src/schema/associations.ts

```diff
@@ -123,33 +146,37 @@ export async function buildAssociations(
         rule.target.suffix,
         rule.target?.entities ?? [],
       ).next().value
-      if (Array.isArray(file)) {
-        file = file[0]
-      }
-    } catch (error) {
-      if (
-        error && typeof error === 'object' && 'code' in error &&
-        error.code === 'MULTIPLE_INHERITABLE_FILES'
-      ) {
-        // @ts-expect-error
+    } catch (error: any) {
+      if (error?.code === 'MULTIPLE_INHERITABLE_FILES') {
         context.dataset.issues.add(error)
-        break
+        continue
```

Comment:
I *think* we didn't really want to stop building associations on the first one that found `MULTIPLE_INHERITABLE_FILES`. With the `break`, the order of loading associations changes the behavior of the validator; with `continue`, it should be consistent, regardless.

This can also be a separate PR.